code/modules/hydroponics/grown.dm

```diff
@@ -190,3 +191,17 @@
 	else
 		return ..()
 
+/obj/item/reagent_containers/food/snacks/grown/proc/log_action(mob/user, atom/target, what_done)
+	var/reagent_str = reagents.log_list()
+	var/genes_str
+	if(seed && length(seed.genes))
+		var/list/plant_gene_names = list()
+		for(var/datum/plant_gene/trait/T in seed.genes)
+			if(T.dangerous)
+				plant_gene_names += T.name
+		genes_str = english_list(plant_gene_names)
+	else
+		genes_str = "No genes"
```

Comment:
Not sure how if this will make much of a difference, but if you assign a value to this var from the get go, you can avoid doing an `else` later on:
```js
var/genes_str = "No genes"
if(seed && length(seed.genes))
	var/list/plant_gene_names = list()
	for(var/datum/plant_gene/trait/T in seed.genes)
		if(T.dangerous)
			plant_gene_names += T.name
	genes_str = english_list(plant_gene_names)
```
Also this: `for(var/datum/plant_gene/trait/T in seed.genes)`, can probably be made typeless.

Reply:
True that. It'll be a bit more readable.

The typed loop I forgot to take out yeah. Good catch!